### PR TITLE
Extends ping and flood-ping support for mac and unix like systems

### PR DESCRIPTION
**Pull request by aquibbaig**
- Extends ping and flood-ping support for Mac and Unix like systems.
- Fixed a few "dictator-like" code that caused us to explicitly pick substrings from predefined arrays. So made it a bit flexible.

Signed-off-by: Aquib Baig <aquibbaig97@gmail.com>

### CODE REVIEW

File: src/lib/filters/scraps/scrapPing.go [hunk 1]
```diff
@@ -20,7 +21,7 @@ func CLIPingScrap(s *string) (a *utils.TypePingScrap) {
 	l := len(arr)
 	if l > 2 {
 		words := strings.Split(arr[l-2], " ")
-		if words[0] == "rtt" {
+		if words[0] == "rtt" || words[0] == "round-trip" {
```
[the remark this answers is not in the document]
Comment (aquibbaig) [reply src/lib/filters/scraps/scrapPing.go:24]: Yes

File: src/lib/modules/ping/flood_ping.go [hunk 2]
```diff
@@ -154,9 +155,8 @@ func (ps *FloodPing) ping(urlRaw string, packets int, tsdbNameHash string, wg *s
 		wg.Done()
 		return
 	}
-
-	result := *scrap.CLIFLoodPingScrap(resp)
-	newBlock := *tsdb.GetNewBlock("flood-ping", getNormalizedBlockStringFlood(result))
+	result := scrap.CLIFLoodPingScrap(resp)
```
[the remark this answers is not in the document]
Comment (aquibbaig) [reply src/lib/modules/ping/flood_ping.go:158]: Why add a pointer to a module? Makes no sense.

File: src/lib/filters/scraps/scrapPing.go [hunk 3]
```diff
@@ -36,17 +37,42 @@ func CLIPingScrap(s *string) (a *utils.TypePingScrap) {
...
+	// "10 packets transmitted, 2 packets received, 80.0% packet loss"
+	var packetLossLine string = ""
+	// PingValues line is of the form:
+	// round-trip min/avg/max/stddev = 91.299/96.281/101.264/4.983 ms (Mac OS/Unix like systems)
+	// rtt min/avg/max/stddev = 91.299/96.281/101.264/4.983 ms (Linux)
+	var pingValuesLine string = ""
+	// The results for flood ping outputs
+	// are different across different platforms.
+	// So, it was not wise enough to declare an
+	// array index, but rather find the line of occurence.
```
Comment (Harkishen-Singh) [inline src/lib/filters/scraps/scrapPing.go:50]: Kindly change these to

```
// Packetloss line is of the form:
        var (
	       // "10 packets transmitted, 2 packets received, 80.0% packet loss"
	       packetLossLine string
	       // PingValues line is of the form:
	       // round-trip min/avg/max/stddev = 91.299/96.281/101.264/4.983 ms (Mac OS/Unix like systems)
	       // rtt min/avg/max/stddev = 91.299/96.281/101.264/4.983 ms (Linux)
	       pingValuesLine string
)
```

File: src/lib/filters/scraps/scrapPing.go [hunk 4]
```diff
@@ -36,17 +37,42 @@ func CLIPingScrap(s *string) (a *utils.TypePingScrap) {
...
+			pingValuesLine = line
+		} else if strings.Contains(line, "rtt") {
+			pingValuesLine = line
+		}
+	}
+	// This error handler is for an optional security
+	// if we could not find suitable parameters in the
+	// ping command's output.
+	if packetLossLine == "" || pingValuesLine == "" {
+		panic("Couldn't find a suitable response which contains packet loss")
```
Comment (Harkishen-Singh) [inline src/lib/filters/scraps/scrapPing.go:64]: Do we really need to stop bench-routes here? If not, please consider using `logger.Terminal("p", "couldn't find a suitable response which contains packet loss")`.

Also please note that error lines should not contain any capital letters.

File: src/lib/modules/ping/flood_ping.go (hunk 2 above)
Comment (Harkishen-Singh) [inline src/lib/modules/ping/flood_ping.go:158]: I got the reason why it was used as value `*`. 

The returned value is an address, so it should rather be as `*(scrap.CLIFLoodPingScrap(resp))` if we want it was done previously. Else, we can go with the current change as well. I don't think there will be any change in performance.

File: src/lib/utils/http.go [hunk 5]
```diff
@@ -24,9 +24,8 @@ const (
 // Specifying addresses directly speeds the entire process manyfolds.
 func CLIPing(url string, packets int) (*string, error) {
 	url = *filters.HTTPPingFilter(&url)
-	cmd := exec.Command(CmdPingBasedOnPacketsNumber, "-c", strconv.Itoa(packets), url)
-	cmdOutput, err := cmd.CombinedOutput()
-	cmdStr := string(cmdOutput)
+	cmd, err := exec.Command(CmdPingBasedOnPacketsNumber, "-c", strconv.Itoa(packets), url).Output()
```
Comment (Harkishen-Singh) [inline src/lib/utils/http.go:27]: why aren't we checking the `err`?

File: src/lib/modules/ping/flood_ping.go (hunk 2 above)
Comment (aquibbaig) [reply src/lib/modules/ping/flood_ping.go:158]: We go in as per the current

**Comment by Harkishen-Singh** [comment]
@aquibbaig , can you enable travis on mac? If that goes well, https://github.com/zairza-cetb/bench-routes/issues/144 will then be fixed as well.

**Comment by Harkishen-Singh** [comment]
Ahh... i see them failing again. Better to remove mac again for now.

**Comment by Harkishen-Singh** [comment]
Also, consider pulling the latest commits for semaphoreci to run.

**Pull request merged**